src/textual/widgets/_option_list.py

```diff
@@ -558,6 +555,38 @@ def _refresh_content_tracking(self, force: bool = False) -> None:
         # list, set the virtual size.
         self.virtual_size = Size(self.scrollable_content_region.width, len(self._lines))
 
+    def _duplicate_id_check(self, candidate_items: list[OptionListContent]) -> None:
+        """Check the items to be added for any duplicates.
+
+        Args:
+            candidate_items: The items that are going be added.
+
+        Raises:
+            DuplicateID: If there is an attempt to use a duplicate ID.
+        """
+        # Get all of the incoming IDs.
+        new_option_ids = [
+            item.id
+            for item in candidate_items
+            if isinstance(item, Option) and item.id is not None
+        ]
+        # Check for duplicates of already-known IDs and also duplicates
+        # within the list of incoming IDs.
+        current_ids = self._option_ids
+        duplicates = {
+            item_id for item_id in new_option_ids if item_id in current_ids
+        }.union(
+            item_id
+            for item_id, id_count in Counter(new_option_ids).items()
+            if id_count > 1
+        )
+        # If there are any duplicates...
+        if duplicates:
+            # ...complain.
+            raise DuplicateID(
```

Comment:
Sorry to nit, but I think the code is self-evident without the comments?

Reply:
Eh, I don't much care for commented code where the comments fizzle out. Feel free to PR to the PR to taste though.

Reply:
Perhaps I should just stay in my lane and not comment on PR's in future. 

I'm just a minor contributor so obviously you can take or leave my suggestions.

Reply:
Nah, most welcome. I actually tried to add you to the reviewer list on this one when I opened the PR but apparently GitHub wasn't having it. 🤷🏻‍♀️